src/daemon/XorgDisplayServer.cpp

```diff
@@ -233,6 +233,23 @@ namespace SDDM {
         // log message
         qDebug() << "Display server stopped.";
 
+        QString displayStopCommand = mainConfig.XDisplay.DisplayStopCommand.get();
+
+        // create display setup script process
+        QProcess *displayStopScript = new QProcess();
+
+        // set process environment
+        QProcessEnvironment env;
+        env.insert("DISPLAY", m_display);
+        env.insert("HOME", "/");
+        env.insert("PATH", mainConfig.Users.DefaultPath.get());
+        env.insert("SHELL", "/bin/sh");
+        displayStopScript->setProcessEnvironment(env);
+
+        // start display setup script
+        qDebug() << "Running display stop script " << displayStopCommand;
+        displayStopScript->start(displayStopCommand);
```

Comment:
that won't work because then you're deleting the displayScript startup script potentially before it's finished.

in this case it kinda leaked for a reason... though a terrible one.

you'd need something like
connect(process, finished, process, deleteLater())); 

or use QProcess::startDetatched which does it all internally.